src/SSVOpenHexagon/Core/HexagonGame.cpp

```diff
@@ -537,6 +537,17 @@ void HexagonGame::playPackSoundOverride(
 void HexagonGame::newGame(const std::string& mPackId, const std::string& mId,
     bool mFirstPlay, float mDifficultyMult, bool executeLastReplay)
 {
+    // Save replay when restarting without having died
+    if(!mFirstPlay && shouldSaveScore() && !status.hasDied)
+    {
+        (void)death_saveScoreIfNeeded(); // Saves local best
+
+        const replay_file rf = death_createReplayFile();
+
+        ssvu::lo("Replay") << "Attempting to send and save replay...\n";
+        death_sendAndSaveReplay(rf);
```

Comment:
I'll do that, but keep in mind that there is still slightly different replay saving logic being called in the death function, so it still needs to be updated in two places in case you wanna change anything.

Reply:
I am aware, unfortunately. I am specifically talking about 

```cpp
(void)death_saveScoreIfNeeded(); // Saves local best

const replay_file rf = death_createReplayFile();

ssvu::lo("Replay") << "Attempting to send and save replay...\n";
death_sendAndSaveReplay(rf);
```

Better to not make the situation worse than it already is :D